### PR TITLE
allocate link directive

### PR DESCRIPTION
**Pull request by kuda1992**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/EUI-3799

### Change description ###
Created a new endpoint which takes the caseId and checks if the user can allocate a role

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: api/workAllocation2/roleService.ts [hunk 1]
```diff
@@ -12,3 +13,15 @@ export async function handleGetRolesByCaseId(path: string, req: EnhancedRequest)
   const headers = setHeaders(req);
   return await httpMock.get<CaseRole>(path, {headers});
 }
+
+export function handleShowAllocatorLinkByCaseId(caseId: string, req: EnhancedRequest): boolean {
```
Comment (udaydenduluri33) [inline api/workAllocation2/roleService.ts:17]: @kuda1992  I can see you looking for caseId below. But, you also need to look if the the user has the case allocator role. That is missing. Could you please add it? Thanks

File: api/workAllocation2/interfaces/caseRole.ts [hunk 2]
```diff
@@ -12,3 +12,10 @@ export interface Action {
   id: string;
   title: string;
 }
+
+export enum RoleType {
+  LEAD_JUDGE = 'Lead Judge',
```
Comment (rayliangatos) [inline api/workAllocation2/interfaces/caseRole.ts:17]: there has been a similar TypeOfRole in src\role-access\models\allocate-role.enum

File: src/cases/directives/show-allocate-link.directive.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  @Input() public showAllocateRoleLink: boolean = false;
+  @Input() public roles: CaseRole[];
+  @Input() public userType: string;
+
+  constructor(private element: ElementRef<HTMLElement>) {
+  }
+
+  private static canDisplayLink(roles: CaseRole[], userType: string, showAllocateRoleLink: boolean): boolean {
+    const show = false;
+    if (!showAllocateRoleLink || userType === 'legalOps' && ShowAllocateLinkDirective.hasExceededNumberCaseManagerRoles(roles)) {
```
Comment (connorpgpmcelroy) [inline src/cases/directives/show-allocate-link.directive.ts:18]: 'legalOps' should be a constant/enum along with 'judiciary' so if we need to change any mention of it, looking through code is not necessary

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Further to Uday's comment, would prefer legalops and judiciary in a constants/enum file

File: src/cases/directives/show-allocate-link.directive.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import { Directive, ElementRef, Input, OnInit } from '@angular/core';
+import { CaseRole } from 'api/workAllocation2/interfaces/caseRole';
+import { TypeOfRole } from 'src/role-access/models';
```
Comment (udaydenduluri33) [inline src/cases/directives/show-allocate-link.directive.ts:3]: please move src to relative path

File: src/cases/components/roles-and-access/roles-and-access.component.ts [hunk 5]
```diff
@@ -1,13 +1,17 @@
 import { Component, Input } from '@angular/core';
 import { CaseView } from '@hmcts/ccd-case-ui-toolkit';
-import { CaseRole } from '../../../../api/workAllocation2/interfaces/caseRole';
 import { LocationInfo } from 'src/app/store/reducers/app-config.reducer';
```
Comment (udaydenduluri33) [inline src/cases/components/roles-and-access/roles-and-access.component.ts:3]: please move src to relative path

File: src/cases/directives/show-allocate-link.directive.spec.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { TypeOfRole } from 'src/role-access/models';
```
Comment (udaydenduluri33) [inline src/cases/directives/show-allocate-link.directive.spec.ts:1]: please move src to relative path

File: src/cases/resolvers/case-roles-resolver.service.spec.ts [hunk 7]
```diff
@@ -4,17 +4,15 @@ import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { TestBed } from '@angular/core/testing';
 import { ActivatedRouteSnapshot } from '@angular/router';
 import { RouterTestingModule } from '@angular/router/testing';
-import { StoreModule } from '@ngrx/store';
+import { of } from 'rxjs';
+import { TypeOfRole } from 'src/role-access/models';
```
Comment (udaydenduluri33) [inline src/cases/resolvers/case-roles-resolver.service.spec.ts:8]: please move src to relative path

File: api/workAllocation2/constants/roles.mock.data.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
+import { TypeOfRole } from '../../../src/role-access/models';
```
Comment (connorpgpmcelroy) [inline api/workAllocation2/constants/roles.mock.data.ts:1]: import contains 'src'

File: api/workAllocation2/util.ts [hunk 9]
```diff
@@ -206,8 +206,12 @@ export function applySearchFilter(person: Person, domain: string, searchTerm: an
...
   const headers = setHeaders(req);
-  const response: AxiosResponse = await http.post(path, payload, { headers });
+  const response: AxiosResponse = await http.post(path, payload, {headers});
   // Return the whole response, not just the data, so we can
   // see what the status of the response is.
   return response;
 }
+
+export function hasCaseAllocatorRole(authorisations: string[]): boolean {
+  return authorisations.some(authorisation => authorisation === 'case-allocator-role');
```
Comment (connorpgpmcelroy) [inline api/workAllocation2/util.ts:216]: Think 'case-allocator-role' may already be a constant within the application so we should use that

File: src/role-access/components/case-roles-table/case-roles-table.component.html [hunk 10]
```diff
@@ -36,7 +36,7 @@
 <dl class="govuk-summary-list" *ngIf="!roles ||  !roles.length">
   <div class="govuk-summary-list__row">
     <dd class="govuk-summary-list__value">
-      There are no {{roleType}} roles for this case.
+      There are no {{userType === 'legalOps' ? 'legal ops' : userType }} roles for this case.
```
Comment (connorpgpmcelroy) [inline src/role-access/components/case-roles-table/case-roles-table.component.html:39]: Could we not use the enum here?

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Minor changes required

File: api/workAllocation2/util.ts (hunk 9 above)
Comment (connorpgpmcelroy) [reply api/workAllocation2/util.ts:216]: checked and it's a constant in app.constants.ts

Comment (connorpgpmcelroy) [reply api/workAllocation2/util.ts:216]: and also in api/user/constants.ts

**Review by udaydenduluri33 — APPROVED** [review]
Approving after the changes requested

**Review by connorpgpmcelroy — APPROVED** [review]
Comments have been discussed/addressed and functionality looks good

**Pull request merged**